### PR TITLE
Add tests selection menu.

### PR DESCRIPTION
**Pull request by agrojean-ledger**
* Add tests selection menu
* Update packages :
    * Update ESLint to latest version :
        * Use flat config file (js syntax),
        * Update precommit hooks,
        * Indent/format/lint ts/js files following latest ESLint rules.
        * Rewrite index.ts test file to match latest glob package syntax
* Optimize task activation : 
    * use only one `forEach` loop in `checkDisabledTasks`
    * Allow to regenerate a subset of tasks with `regenerateSubset`

Tested on the following platforms : 
- [x] Linux
- [x] Windows
- [x] macOS

Tested on the following apps:
- [x] app-boilerplate
- [x] app-zilliqa
- [x] app-bitcoin-new
- [x] app-radix-babylon

### CODE REVIEW

File: src/taskProvider.ts [hunk 1]
```diff
@@ -555,34 +601,70 @@ export class TaskProvider implements vscode.TaskProvider {
...
     // Runs functional tests inside the docker container (with Qt display enabled).
     const exec = `docker exec -it ${this.containerName} bash -c 'pytest ${
       this.functionalTestsDir
-    } --tb=short -v --device ${this.tgtSelector.getSelectedSpeculosModel()} --display'`;
+    } --tb=short -v --device ${this.tgtSelector.getSelectedSpeculosModel()} --display ${testsSelection}'`;
     return exec;
   }
 
   private functionalTestsGoldenRunExec(): string {
+    let testsSelection = "";
```
Comment (cedelavergne-ledger) [inline src/taskProvider.ts:638]: This bloc is written several times and should be factorize in a function like `setSelectedTests`), which returns an empty string of the full filter `"-k ... or ..."`

File: src/appSelector.ts [hunk 2]
```diff
@@ -518,6 +606,102 @@ function getAppVariants(appdir: string, appName: string, folderUri: vscode.Uri):
...
+    // Find the tests list with pytest :
+    // * Get the device option from pytest help (either --model or --device)
+    // * If the device option is found, run pytest with --collect-only and the device option
+    // * If the device option is not found, run pytest with --collect-only
+    let getTestsListCmd = "docker";
+
+    const varPrefix = process.platform === "win32" ? "\`$" : "$";
+    const quotesAroundBashCommand = process.platform === "win32" ? "\"" : "";
+    let getTestsListArgs = [
+      "exec", "-u", "0", selectedApp!.containerName, "bash", "-c",
```
Comment (cedelavergne-ledger) [inline src/appSelector.ts:639]: Do we really need to start in `root`?

**Review by cedelavergne-ledger — CHANGES_REQUESTED** [review]

**Comment by cedelavergne-ledger** [comment]
Missing the information in the UI that a test selection has been performed, especially because they are stored in the settings.
Should be interesting to modify the UI entry with something like `Run tests [Selected only]`

File: src/appSelector.ts (hunk 2 above)
Comment (agrojean-ledger) [reply src/appSelector.ts:639]: Yes to install the requirements with `pip install -r requirements.txt`, if we don't install the requirements, tests collection with `pytest --collect-only` might fail (as pytest needs to execute the tests files to collect the tests, even if not executing the tests themselves)

File: src/taskProvider.ts (hunk 1 above)
Comment (agrojean-ledger) [reply src/taskProvider.ts:638]: Ok, adding a function.

File: src/appSelector.ts [hunk 3]
```diff
@@ -518,6 +606,102 @@ function getAppVariants(appdir: string, appName: string, folderUri: vscode.Uri):
...
+      `${quotesAroundBashCommand}cd ${selectedApp.functionalTestsDir} &&
+        pip install -r requirements.txt > /dev/null 2>&1 &&
+        clear &&
+        device_option=${varPrefix}(pytest --help |
```
[the remark this answers is not in the document]
Comment (agrojean-ledger) [reply src/appSelector.ts:646]: It works as is, I don't remember which corner cases are covered by the current version but I'd prefer not modifying it.

**Comment by agrojean-ledger** [comment]
> Missing the information in the UI that a test selection has been performed, especially because they are stored in the settings. Should be interesting to modify the UI entry with something like `Run tests [Selected only]`

I've added a visual indicator in the "Functional tests" tree view item.

**Review by cedelavergne-ledger — APPROVED** [review]
